docs/Secure-Coding-Guide-for-Python/CWE-707/CWE-78/README.md

```diff
@@ -0,0 +1,215 @@
+|Get user/group id|id|`os.getuid()`<br>`os.setuid()`|
+|Get OS and/or kernel type and name|uname -as|`os.uname()`|
+
+<sub>table 00, example list of Unix commands and their Python equivalents.</sub>
+
+Any variation of using input from a lesser trusted source as part of a command line program has a very high probability of resulting in a potential attack including the use of specialized modules. Consider:
+
+* *CWE-184: Incomplete List of Disallowed Input.*
+* *CWE-209: Generation of Error Message Containing Sensitive Information.*
+* *[CWE-501: Trust Boundary Violation](https://github.com/ossf/wg-best-practices-os-developers/blob/main/docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-501/README.md)*
+
+## Non-Compliant Code Example (Read Only)
+
+This scenario demonstrates a potential remote command execution. The `FileOperations.list_dir()` method allows an attacker to inject commands into the string dirname such as `head -1 /etc/passwd` under Linux or `net user` under Windows. Note that older versions of `Python < 3.9.12` allowed to turn a non-interactive shell into an active shell in Windows by providing `cmd.exe /C` as an argument  [[python.org 3.12.5 - Subprocess management]](https://docs.python.org/3/library/subprocess.html).
+
+*[noncompliant01.py](noncompliant01.py):*
+
+```python
+""" Non-compliant Code Example """
+# SPDX-FileCopyrightText: OpenSSF project contributors
+# SPDX-License-Identifier: MIT
+""" Non-compliant Code Example """
+from subprocess import Popen
+import os
+
+
+class FileOperations:
+    """Helper class for file system operations"""
+
+    def list_dir(self, dirname: str):
+        """List the contents of a directory"""
+        if "nt" in os.name:
+            Popen("dir " + dirname, shell=True).communicate()
+        if "posix" in os.name:
+            Popen("ls " + dirname, shell=True).communicate()
+
+
+#####################
+# Trying to exploit above code example
+#####################
+if "nt" in os.name:
+    FileOperations().list_dir("%HOMEPATH% & net user")
+if "posix" in os.name:
+    FileOperations().list_dir("/etc/shadow; head -1 /etc/passwd")
+
+```
+
+Instead of listing files, the code in `noncompliant01.py` prints the first line of `/etc/passwd` on Linux or starts `net user` under Windows.
+
+## Non-Compliant Code Example (Read, Write)
+
+The attack surface increases if a user is also allowed to upload or create files or folders.
+
+The `noncompliant02.py` example demonstrates the injection via file or folder name that is created prior to using the `list_dir()` method. We assume here that an untrusted user is allowed to create files or folders named `& calc.exe or ;ps aux` as part of another service such as upload area, submit form, or as a result of a zip-bomb as per *CWE-409: Improper Handling of Highly Compressed Data (Data Amplification)*. Encoding issues as described in *CWE-180: Incorrect Behavior Order: Validate Before Canonicalize* must also be considered.
```

Comment:
Maybe there should be a link to this?

Reply:
As discussed, links will come once those risks have a readme